pages/spacedeck/[[...roomId]].js

```diff
@@ -149,20 +150,48 @@ export default function Spacedeck() {
     }, [spacedeck]);
 
     async function createSketchRoom(link, name, parent = serviceSpaceId) {
-        // eslint-disable-next-line no-undef
-        logger.debug('creating room for ' + name);
-        const room = await matrix.createRoom(name, false, '', 'invite', 'content', 'spacedeck').catch(() => {
-            setErrorMessage(t('Something went wrong when trying to create a new room'));
-        });
-        await auth.getAuthenticationProvider('matrix').addSpaceChild(parent, room).catch(() => {
-            setErrorMessage(t('Couldn\'t add the new room to your sketch folder'));
-        });
-        await matrixClient.sendMessage(room, {
-            msgtype: 'm.text',
-            body: link,
-        }).catch(() => {
-            setErrorMessage(t('Something went wrong when trying to save the new sketch link'));
-        });
+        // Log debugging information
+        setUserFeedback(t('Syncing {{name}} from server', { name: name }));
+
+        // Create the room with retry handling
+        const createRoomForSketch = async () => {
+            return await matrix.createRoom(name, false, '', 'invite', 'content', 'spacedeck');
+        };
+
+        const room = await createRoomForSketch()
+            .catch(async (error) => {
+                return matrix.handleRateLimit(error, () => createRoomForSketch())
+                    .catch(error => setErrorMessage(error.message));
+            });
+
+        // Log debug information about current progress
+        logger.debug(`Created room for ${name} with id ${room}`);
+
+        // Add the room as a child to the parent space
+        const addSpaceChild = async () => await auth.getAuthenticationProvider('matrix').addSpaceChild(parent, room);
+
+        await addSpaceChild()
+            .catch(async (error) => {
+                return matrix.handleRateLimit(error, () => addSpaceChild())
+                    .catch(error => setErrorMessage(error.message));
+            });
+
+        // Log debug information about current progress
+        logger.debug('Added %s to parent %s', name, parent);
+
+        // Send the message to the room with retry handling
+        const sendMessage = async () => {
+            await matrixClient.sendMessage(room, {
+                msgtype: 'm.text',
+                body: link,
+            });
+        };
+
+        await sendMessage()
+            .catch(async (error) => {
+                return matrix.handleRateLimit(error, () => sendMessage())
```

Comment:
So basically you've been adding the very same error handling in two different places in the codebase.
Both -- here in /sketch and also in /write -- we add a space child, and we add the very same retry handler, if something goes wrong. Can't we just do that inside `addSpaceChild()` in `Matrix.js` then?

I'd even say we _should_ do so.

And this also counts for `.createRoom()` for example. Why add a retry handler in all the places that create rooms, when you could just do it in `Matrix.js`?